### PR TITLE
feat(react): ✨ add types for react-typical

### PR DESCRIPTION
**Pull request by Ankan002**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/react-typical/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        "noImplicitThis": true,
+        "strictFunctionTypes": true,
+        "strictNullChecks": true,
+        "baseUrl": "../",
+        "typeRoots": ["../"],
+        "types": [],
+        "noEmit": true,
+        "forceConsistentCasingInFileNames": true,
+        "jsx": "react",
+        "esModuleInterop": true
```
Comment (peterblazejewicz) [inline types/react-typical/tsconfig.json:15]:
```suggestion
        "jsx": "react"
```
why you would need this here?
https://github.com/DefinitelyTyped/DefinitelyTyped#esmoduleinteropallowsyntheticdefaultimports

File: types/react-typical/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+// Type definitions for react-typical 0.1
+// Project: https://github.com/catalinmiron/react-typical#readme
+// Definitions by: Ankan Bhattacharya <https://github.com/Ankan002>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import { ReactElement } from "react";
+
+export interface ReactTypicalProps {
+    steps: Array<string|number>;
```
Comment (peterblazejewicz) [inline types/react-typical/index.d.ts:9]:
```suggestion
    steps: Array<string | number>;
```

File: types/react-typical/tsconfig.json (hunk 1 above)
Comment (Ankan002) [reply types/react-typical/tsconfig.json]: @peterblazejewicz actually this enabled me to use ```import React from "react"```. But now I removed it and I am using ```import * as React from "react"```.

**Comment by Ankan002** [comment]
@peterblazejewicz Please review the changes as I implemented the suggestions that you have given me before.

**Review by sandersn — APPROVED** [review]

**Pull request merged**